src/weather.cpp

```diff
@@ -66,15 +67,122 @@ static bool is_player_outside()
     return g->m.is_outside( point( g->u.posx(), g->u.posy() ) ) && g->get_levz() >= 0;
 }
 
-static constexpr int THUNDER_CHANCE = 50;
-static constexpr int LIGHTNING_CHANCE = 600;
+const weather_datum &weather::data( const weather_type type )
+{
+    const size_t i = static_cast<size_t>( type );
+    if( i < weather_datums.size() ) {
+        return weather_datums[i];
+    }
+    debugmsg( "Invalid weather requested." );
+    return weather_datums[WEATHER_NULL];
+}
+
+weather_type weather::get_bad_weather()
+{
+    weather_type bad_weather = WEATHER_NULL;
+    int current_weather = 0;
+    for( const weather_datum &weather : weather_datums ) {
+        current_weather++;
+        if( weather.precip == precip_class::HEAVY ) {
+            bad_weather = current_weather;
+        }
+    }
+    return bad_weather;
+}
+
+int weather::get_count()
+{
+    return weather_datums.size();
+}
+
+void weather::load_weather_type( const JsonObject &jo )
```

Comment:
You should use generic factory, not directly call the json functions.

Reply:
It is mostly done in that branch - https://github.com/CleverRaven/Cataclysm-DDA/compare/master...ZhilkinSerg:jsonize-weather-type#diff-0734d41ef72f36d830002abb8f38db8d